impeller/aiks/color_source.h

```diff
@@ -25,6 +25,90 @@ namespace impeller {
 
 struct Paint;
 
+struct LinearGradientData {
```

Comment:
If we had `Clone()` calls for these contents, you wouldn't have to duplicate the data they need in structs.  So the variant would just be std::variant<LinearGradientContents, RadialGradientContents, ...>.

GetContents would then just call Clone(), but storing it in a variant would allow you to visit the actual type.

Just a suggestion, feel free to punt since this is good.

Reply:
Oh, so rather than have the data at all, just make color source contents cloneable. That makes sense but would also require more refactors, since we store this on paint and do a lot of type based dispatching.

Reply:
Though we could also change ColorSourceContents to remove the inheritance and change its behavior based on the variant its given.